progressutil/iocopy.go

```diff
@@ -66,21 +53,36 @@ func (cr *copyReader) updateProgressBar() error {
 	return cr.pb.SetCurrentProgress(progress)
 }
 
+// NewCopyProgressPrinter returns a new CopyProgressPrinter
+func NewCopyProgressPrinter() *CopyProgressPrinter {
+	return &CopyProgressPrinter{results: make(chan error), cancel: make(chan struct{})}
+}
+
 // CopyProgressPrinter will perform an arbitrary number of io.Copy calls, while
 // continually printing the progress of each copy.
 type CopyProgressPrinter struct {
-	readers []*copyReader
-	errors  []error
+	results chan error
+	cancel  chan struct{}
+
 	lock    sync.Mutex
```

Comment:
@s-urbaniak aiui this is communicated through the struct layout - the lock protects the members listed immediately after it, and if there are multiple locks the struct members are grouped into blocks separated by an empty line. Wdyt?